conans/test/functional/graph_lock/build_order_test.py

```diff
@@ -0,0 +1,507 @@
+import json
+import unittest
+
+from parameterized import parameterized
+
+from conans.test.utils.tools import TestClient, GenConanfile
+from conans.util.env_reader import get_env
+
+
+class BuildOrderTest(unittest.TestCase):
+
+    def single_consumer_test(self):
+        # https://github.com/conan-io/conan/issues/5727
+        client = TestClient()
+        client.save({"conanfile.py": GenConanfile("test4", "0.1")})
+        client.run("lock create conanfile.py --build --lockfile-out=conan.lock")
+        client.run("lock build-order conan.lock --json=bo.json")
+        jsonbo = json.loads(client.load("bo.json"))
+        self.assertEqual([], jsonbo)
+
+    @parameterized.expand([(True,), (False,)])
+    def build_not_locked_test(self, export):
+        # https://github.com/conan-io/conan/issues/5727
+        client = TestClient()
+        client.save({"conanfile.py": GenConanfile("test4", "0.1")})
+        if export:
+            client.run("export .")
+            client.run("lock create --reference=test4/0.1@ --lockfile-out=conan.lock")
+        else:
+            client.run("create .")
+            client.run("lock create --reference=test4/0.1@ --build=test4 --lockfile-out=conan.lock")
+        if client.cache.config.revisions_enabled:
+            ref = "test4/0.1#f876ec9ea0f44cb7adb1588e431b391a"
```

Comment:
Please, do not use `GenConanfile` if using recipe revisions explicitly, **or read the revision dynamically**. It is very annoying to see totally unrelated tests failing when something changes in that class. 

This is extensible to other tests in this class